bot/exts/utils/reminders.py

```diff
@@ -30,6 +32,133 @@
 ReminderMention = t.Union[UnambiguousUser, discord.Role]
 
 
+class SnoozeSelectView(discord.ui.View):
+    """The reminder's select dropdown UI View."""
+
+    SNOOZE_DURATIONS: dict[str, t.Optional[int]] = {
+        # Mapping of `duration string: num_seconds_in_duration`.
+        #
+        # `num_seconds_in_duration` is `None` for durations >= 1 month
+        # since the amount of seconds depends on the date of lookup
+        # (i.e. seconds from 1st jan to 1st feb is different than from 1st feb to 1st march etc).
+        "1 minute": 60,
+        "5 minutes": 5 * 60,
+        "15 minutes": 15 * 60,
+        "30 minutes": 30 * 60,
+        "1 hour": 60 * 60,
+        "3 hours": 3 * 60 * 60,
+        "6 hours": 6 * 60 * 60,
+        "1 day": 24 * 60 * 60,
+        "1 week": 7 * 24 * 60 * 60,
+        "1 month": None,
+        "3 months": None,
+        "6 months": None,
+        "1 year": None
+    }
+
+    def __init__(self, bot: Bot, reminder_id: int, reminder_author_id: int):
+        super().__init__(timeout=30)
+        self.new_expiry: t.Optional[datetime] = None
+        self.reminder_was_snoozed = False
+
+        self.bot = bot
+        self.reminder_id = reminder_id
+        self.interaction_owner_id = reminder_author_id
+
+        self.dropdown: discord.ui.Select = self.children[0]
```

Comment:
Why are we doing this? The defined select can be accessed as `self.select_snooze_duration`.

Reply:
> Why are we doing this? The defined select can be accessed as `self.select_snooze_duration`.

This was simply copied from the code on lancebot:
https://github.com/python-discord/sir-lancebot/blob/main/bot/exts/utilities/epoch.py#L115

Reply:
Upon further inspection, I don't see this? `self.select_snooze_duration` is a method, so doesn't give us the Select menu.

Reply:
It is decorated with the `discord.ui.select` decorator which converts it into a Select.

Reply:
@TizzySaurus I think you accidental resolved this without fixing it.

Reply:
> It is decorated with the `discord.ui.select` decorator which converts it into a Select.

Right, yes, I missed that. Have fixed :+1: